core/EE_Addon.core.php

```diff
@@ -597,7 +597,7 @@ public function set_req_type($req_type)
      */
     public function detect_req_type()
     {
-        if (! $this->_req_type) {
+        if ($this->_req_type === null) {
```

Comment:
the numeric value for a normal request is `0` which would result in calls to this function repeatedly running `detect_activation_or_upgrade()` because `! 0` == `true`